homeassistant/components/trafikverket_train/sensor.py

```diff
@@ -0,0 +1,172 @@
+"""Train information for departures and delays, provided by Trafikverket."""
+
+import asyncio
+from datetime import date, datetime, timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_API_KEY, CONF_NAME, CONF_WEEKDAY, WEEKDAYS
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_TRAINS = "trains"
+CONF_FROM = "from"
+CONF_TO = "to"
+CONF_TIME = "time"
+
+ATTR_CANCELED = "canceled"
+ATTR_DELAY_TIME = "number_of_minutes_delayed"
+ATTR_PLANNED_TIME = "planned_time"
+ATTR_ESTIMATED_TIME = "estimated_time"
+ATTR_ACTUAL_TIME = "actual_time"
+ATTR_OTHER_INFORMATION = "other_information"
+ATTR_DEVIATIONS = "deviations"
+
+ICON = "mdi:train"
+SCAN_INTERVAL = timedelta(minutes=5)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_TRAINS): [{
+        vol.Required(CONF_NAME): cv.string,
+        vol.Required(CONF_TO): cv.string,
+        vol.Required(CONF_FROM): cv.string,
+        vol.Optional(CONF_TIME): cv.time,
+        vol.Optional(CONF_WEEKDAY, default=WEEKDAYS):
+            vol.All(cv.ensure_list, [vol.In(WEEKDAYS)])}]
+})
+
+
+@asyncio.coroutine
+def async_setup_platform(
```

Comment:
We've migrated to `async`/`await` syntax. Please drop this decorator and prefix async methods with `async` like `async def async_setup_platform`. Replace any `yield from` with `await`.